File: sub-core/src/main/java/org/github/spring/bootstrap/AppCtxHolder.java
package org.github.spring.bootstrap;

import lombok.NonNull;
import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.stereotype.Component;

@Component
public class AppCtxHolder implements ApplicationContextAware {
    private static ApplicationContext appCtx;

    public static @NonNull ApplicationContext getAppCtx() {
        return appCtx;
    }

    @Override
    public void setApplicationContext(@NonNull ApplicationContext context) throws BeansException {
        appCtx = context;
    }
}

File: sub-core/src/main/java/org/github/spring/bootstrap/AppCtxHolder.kt
package org.github.spring.bootstrap

import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.stereotype.Component

/**
 * ApplicationContext Holder.
 *
 * @author JYD_XL
 */
@Component
class AppCtxHolder: ApplicationContextAware {
  override fun setApplicationContext(applicationContext: ApplicationContext) {
    context = applicationContext
  }

  companion object {
    /** GET ApplicationContext.  */
    fun getAppCtx() = context

    /** ApplicationContext.  */
    private lateinit var context: ApplicationContext
  }
}
